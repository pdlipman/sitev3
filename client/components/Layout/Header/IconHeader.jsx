import React from 'react';
import PropTypes from 'prop-types';

import { Header, Icon } from 'semantic-ui-react';

export default class IconHeader extends React.Component {
    static propTypes = {
        contents: PropTypes.string.isRequired,
        headerTag: PropTypes.string,
        iconName: PropTypes.string.isRequired,
        subHeader: PropTypes.string,
    };

    static defaultProps = {
        headerTag: 'h2',
        subHeader: '',
    };

    render() {
        const {
            contents,
            headerTag,
            iconName,
            subHeader,
        } = this.props;

        return (
            <Header
                as={headerTag}
            >
                <Icon
                    name={iconName}
                />
                <Header.Content>
                    { contents }
                    <Header.Subheader>
                        { subHeader }
                    </Header.Subheader>
                </Header.Content>
            </Header>
        );
    }
}
