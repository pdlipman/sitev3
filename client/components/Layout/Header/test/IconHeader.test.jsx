import React from 'react';
import { shallow } from 'enzyme';

import IconHeader from '../IconHeader';

function shallowWrapper() {
    return shallow(
        <IconHeader
            contents='Test Contents'
            iconName='home'
        />,
    );
}

describe('IconHeader.jsx', () => {
    test('IconHeader should render properly', () => {
        const wrapper = shallowWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    test('IconHeader render should throw error', () => {
        expect(() => {
            shallow(<IconHeader />);
        }).toMatchSnapshot();
    });
});
