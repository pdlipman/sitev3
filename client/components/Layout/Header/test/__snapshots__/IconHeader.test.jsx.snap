// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconHeader.jsx IconHeader render should throw error 1`] = `[Function]`;

exports[`IconHeader.jsx IconHeader should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconHeader
    contents="Test Contents"
    headerTag="h2"
    iconName="home"
    subHeader=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "h2",
      "children": Array [
        <Icon
          as="i"
          name="home"
        />,
        <HeaderContent>
          Test Contents
          <HeaderSubheader>
            
          </HeaderSubheader>
        </HeaderContent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "name": "home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Test Contents",
            <HeaderSubheader>
              
            </HeaderSubheader>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Test Contents",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "h2",
        "children": Array [
          <Icon
            as="i"
            name="home"
          />,
          <HeaderContent>
            Test Contents
            <HeaderSubheader>
              
            </HeaderSubheader>
          </HeaderContent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "name": "home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Test Contents",
              <HeaderSubheader>
                
              </HeaderSubheader>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Test Contents",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
